src/test/java/usociety/manager/domain/service/user/impl/CreateUserDelegateImplTest.java

```diff
@@ -0,0 +1,168 @@
+package usociety.manager.domain.service.user.impl;
+
+import static java.lang.Boolean.FALSE;
+import static java.lang.Boolean.TRUE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+import static usociety.manager.app.api.UserApiFixture.email;
+import static usociety.manager.app.api.UserApiFixture.username;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InOrder;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import usociety.manager.app.api.UserApi;
+import usociety.manager.app.api.UserApiFixture;
+import usociety.manager.app.rest.request.CreateUserRequest;
+import usociety.manager.domain.exception.GenericException;
+import usociety.manager.domain.exception.WebException;
+import usociety.manager.domain.provider.user.UserConnector;
+import usociety.manager.domain.provider.user.dto.UserDTO;
+import usociety.manager.domain.service.common.CloudStorageService;
+import usociety.manager.domain.service.email.MailService;
+import usociety.manager.domain.service.otp.OtpService;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CreateUserDelegateImplTest {
+
+    private static final String OTP_CODE = "987654";
+
+    @Mock
+    private CloudStorageService cloudStorageService;
+    @Mock
+    private UserConnector userConnector;
+    @Mock
+    private MailService mailService;
+    @Mock
+    private OtpService otpService;
+
+    @InjectMocks
+    private CreateUserDelegateImpl subject;
+
+    private CreateUserRequest request;
+    private UserDTO userDTO;
+    private UserApi userApi;
+
+    @Before
+    public void setUp() {
+        ReflectionTestUtils.setField(subject, "validateOtp", TRUE, Boolean.class);
+
+        request = CreateUserRequest.newBuilder()
+                .password(UserApiFixture.password)
+                .photo("base64Image")
+                .username(username)
+                .otpCode(OTP_CODE)
+                .name("John Doe")
+                .email(email)
+                .build();
+
+        userDTO = UserDTO.newBuilder()
+                .username(username)
+                .email(email)
+                .name(UserApiFixture.name)
+                .id(UserApiFixture.id)
+                .build();
+
+        userApi = UserApi.newBuilder()
+                .username(username)
+                .email(email)
+                .name(UserApiFixture.name)
+                .id(UserApiFixture.id)
+                .build();
+    }
+
+    @Test
+    public void shouldCreateUserCorrectly() throws GenericException {
```

Comment:
Given

When

Then


https://blog.codecentric.de/en/2017/09/given-when-then-in-junit-tests/

Reply:
Get it rid hahaha

Reply:
Do you have sight problems? I wrote that article hahaha